nvme_gw_server.py

```diff
@@ -62,6 +64,19 @@ def __init__(self, nvme_config):
         self.server = None
         self.transports = set()
 
+        self.gateway_name = self.nvme_config.get("config", "gateway_name")
```

Comment:
We need to make sure `gateway_name` does not have `_` otherwise we probably have a problem with the persistent config. Generally, we might need to check that it only has characters that are valid in an OMAP key. We can let it fail later but than we run into the same semantics again that it fails late.

Reply:
Or use some other char instead of `_` as a separator. We have an issue for this: https://github.com/ceph/ceph-nvmeof/issues/10

Reply:
Ok yes, let's try to address this sooner than later. Otherwise it might be a mess to change. But I agree should be generic to all cases in another PR.